### PR TITLE
new healing system | making it easier to organize

### PR DESCRIPTION
**Pull request by DRuppFv**
I though the fact that you use the healing item in the same way as a weapon a little weird, then, i made these mods to make it "normal".

If it was the intention, I'm sorry, but I don't see any motive to refuse this PR considering that it makes very easier to organize the future new item kinds when grabbed. As I said, if it was the intention, just tell me and I will make the health items be treated as the throwable items, but i don't think there's any reason to refuse this new way of organization. 

Obs: first, I tried to make it possible grab a health even with another item in the inventory, because I forgot that grab and throw are in the same keybind. I think it would be great, but first, these two actions should be divided, so, i have a question: If i divide them, would the PR be accepted?

### CODE REVIEW

**Comment by DRuppFv** [comment]
I didn't know how to format dude :skull:

File: src/fighter_state.rs
Comment (zicklag) [inline src/fighter_state.rs:976]:
```suggestion
                    panic!("Health items should be used immediately, and can't be thrown");
```

I think it'd be better to take out the health logic here and panic if this happens to indicate that this shouldn't happen.

**Review by zicklag — CHANGES_REQUESTED** [review]
This looks good to me, other than a small tweak.

I think it's a good idea, but @erlend-sh and @odecay are the authorities on gameplay I think so I'd like to get a quick approval from one of them before merging.

File: src/fighter_state.rs
Comment (DRuppFv) [reply src/fighter_state.rs:976]: You are completely right. I don't know how i haven't thought on it.

**Comment by odecay** [comment]
The general strategy of differentiating between health items and throwable items seems good for now. May change in the future, for reference think of the Milk Bottle item in LF2. You pick it up, can drink to refill health, and also have the option to throw the bottle.

In terms of inputs for grab and throw, I also think they should be separate inputs, but there has been some disagreement on the subject and even if we were to do it, it will likely come after we upgrade our input handling for more complex cases.
It should not be included in this PR.

**Comment by DRuppFv** [comment]
> The general strategy of differentiating between health items and throwable items seems good for now. May change in the future, for reference think of the Milk Bottle item in LF2. You pick it up, can drink to refill health, and also have the option to throw the bottle.
> 
> In terms of inputs for grab and throw, I also think they should be separate inputs, but there has been some disagreement on the subject and even if we were to do it, it will likely come after we upgrade our input handling for more complex cases. It should not be included in this PR.

Should i close it then?

Obs: Don't care about the commits. I REALLY don't know how did that happen. I was doing a third squash, and, as you can see, i had done two squashs before that, so, i know how to make one, but for some reason 20 commits appeared after that.

**Comment by odecay** [comment]
No don't close. I think this will be good to get in. I was also noticing how it felt unintuitive to pick up/use health items currently and this will be an improvement.

We will get to further changes in the future.

**Comment by zicklag** [comment]
Sorry for the wait, but I'm going to merge this soon, once I get a chance to fix the git history.

**Comment by DRuppFv** [comment]
> Sorry for the wait @DRuppFv , but I'm going to merge this soon, once I get a chance to fix the git history.

Did you read what i said about it? I really don't know how did that happen. If you think it's necessary, i can close this PR and open a new one

**Comment by zicklag** [comment]
> If you think it's necessary, i can close this PR and open a new one

That would work. I was pretty much just going to do the same thing by manually copying your changes and pushing a new commit that includes them, but if you wanted to do it, that would be great!

**Comment by DRuppFv** [comment]
> > If you think it's necessary, i can close this PR and open a new one
> 
> That would work. I was pretty much just going to do the same thing by manually copying your changes and pushing a new commit that includes them, but if you wanted to do it, that would be great!
> 
> I'm not sure what that issue was. Weird... :man_shrugging: 

Ok. I'll try to fix it for the last time and if it don't work, i'll do what i said

**Review by zicklag — APPROVED** [review]
Looks like that worked!

bors r+

**Comment by DRuppFv** [comment]
I had to completely reset the branch but i think it's done.
@zicklag

**Pull request merged**
